osu.Desktop.VisualTests/Tests/TestCaseTaikoHitObjects.cs

```diff
@@ -0,0 +1,164 @@
+            })
+            {
+                Position = new Vector2(100, 300)
+            });
+
+            Add(new RimHitCircle(new AccentedCirclePiece(@"rim")
+            {
+                KiaiMode = kiai
+            })
+            {
+                Position = new Vector2(350, 300)
+            });
+
+            Add(new SwellCircle(new CirclePiece(@"swell")
+            {
+                KiaiMode = kiai
+            })
+            {
+                Position = new Vector2(100, 500)
+            });
+
+            Add(new SwellCircle(new AccentedCirclePiece(@"swell")
+            {
+                KiaiMode = kiai
+            })
+            {
+                Position = new Vector2(350, 500)
+            });
+
+            Add(new DrumRollCircle(new CirclePiece(string.Empty)
+            {
+                Width = 250,
+                KiaiMode = kiai
+            })
+            {
+                Position = new Vector2(575, 100)
+            });
+
+            Add(new DrumRollCircle(new AccentedCirclePiece(string.Empty)
+            {
+                Width = 250,
+                KiaiMode = kiai
+            })
+            {
+                Position = new Vector2(575, 300)
+            });
+        }
+
+        private class SwellCircle : BaseCircle
+        {
+            public SwellCircle(CirclePiece piece)
+                : base(piece)
+            {
+            }
+
+            [BackgroundDependencyLoader]
+            private void load(OsuColour colours)
+            {
+                Piece.AccentColour = colours.YellowDark;
+            }
+        }
+
+        private class DrumRollCircle : BaseCircle
+        {
+            public DrumRollCircle(CirclePiece piece)
+                : base(piece)
+            {
+            }
+
+            [BackgroundDependencyLoader]
+            private void load(OsuColour colours)
+            {
+                Piece.AccentColour = colours.YellowDark;
+            }
+        }
+
+        private class CentreHitCircle : BaseCircle
+        {
+            public CentreHitCircle(CirclePiece piece)
+                : base(piece)
+            {
+            }
+
+            [BackgroundDependencyLoader]
+            private void load(OsuColour colours)
+            {
+                Piece.AccentColour = colours.PinkDarker;
+            }
+        }
+
+        private class RimHitCircle : BaseCircle
+        {
+            public RimHitCircle(CirclePiece piece)
+                : base(piece)
+            {
+            }
+
+            [BackgroundDependencyLoader]
+            private void load(OsuColour colours)
+            {
+                Piece.AccentColour = colours.BlueDarker;
+            }
+        }
+
+        private abstract class BaseCircle : Container
+        {
+            protected readonly CirclePiece Piece;
+
+            protected BaseCircle(CirclePiece piece)
+            {
+                Piece = piece;
+
+                Add(Piece);
+            }
+        }

```

Comment:
Note: These are only here for the purposes of this test case for now, and will be replaced as hit objects are implemented. HitObjects will not be using this BaseCircle class (it's only easier to expose this here for now).